### DISCUSSION TITLE
Installation issue on Windows

### DISCUSSION THREAD

**Discussion by mrssuper** [rabbitmq/rabbitmq-server#2418] [Other] [answered]
Hello
I have a problem installing latest (or previous) version of RabbitMQ on
windows 10, which has latest generation i7 processor. The RabbitMQ service just starts and stops right away (tried on more than 1 pc).
Are there any workarounds/patches available to fix the issue?

Thank you in advance.

**Comment by lukebakken** [comment]
Please tell us _exactly_ how you are installing RabbitMQ:

* What build of Windows 10 are you using?
* Which RabbitMQ package are you installing?
* Which Erlang package are you installing?
* What commands are you running to install RabbitMQ and Erlang?
* Is there a log file in `%AppData%\RabbitMQ`? Please log in as the administrative user you are using to install RabbitMQ, and check this directory. Attach log files to your response.

**Comment by mrssuper** [comment]
- version 1909 (os build 18363.720)
- erlang opt 22 (10.5) , but tried 23 as well
- rabbitmq 3.8, but tried latest as well
- just running windows installer for erlang, then windows installer for rabbitmq server, and then trying to enable management plugin from command line (where it gives a lots of errors since the service isn't running)
- log files doesn't have any info besides one line saying: "Log file opened with Lager"

Some of the other applications we were installing, had issues just because of this processor:
intel core i7-1065g7 10th generation and required a patch.

**Reply by mrssuper** [reply]
tried on 2 brand new laptops HP pavillion 15-cs3019nr, out of the box with nothing else installed on it. same issue. reinstalling windows didn't help as well.

**Comment by michaelklishin** [comment]
RabbitMQ does not have any architecture-specific code. I highly doubt this issue has anything to do with Erlang's compatibility with a particular CPU model since [most common Windows installation issues](https://www.rabbitmq.com/windows-quirks.html) have a lot more boring root causes.

**Comment by mrssuper** [comment]
yet this happens only in this 2 laptops, not anywhere else.

**Comment by michaelklishin** [comment]
@mrssuper that's not a whole lot of evidence. We have years of experience with Windows issues and they usually come down to the usual suspects:

 * Security tools
 * Spaces and non-ASCII characters in RabbitMQ installation paths
 * Multiple Erlang versions installed at the same time

In my 10 year history with this project I don't remember a CPU-specific issue on any OS.

You can try starting an Erlang shell on the command line as `erl.exe` and see if that starts.

There are two other installation methods available on Windows:

 * Chocolatey
 * [Windows binary build](https://www.rabbitmq.com/install-windows-manual.html)

The latter potentially can provide more information to standard output on node start. If all of those mechanisms fail, please start a discussion on the [erlang-questions](http://erlang.org/mailman/listinfo/erlang-questions) mailing list. Our team does not maintain Erlang and unfortunately, there really isn't much information for us to work with.

**Comment by michaelklishin** [comment]
There is also a [community-maintained RabbitMQ Docker image](https://github.com/docker-library/rabbitmq/), which should work with Docker on Windows and sidestep a whole lot of Windows-specific issues since the image uses Ubuntu.

Lastly, our experience with the WSL varied greatly but you can try using that to compare as well.

**Comment by lukebakken** [comment, marked as answer]
> yet this happens only in this 2 laptops, not anywhere else

This is almost certainly environment specific, then. One one of these machines, do the following:

* Log in using the same administrative account you have been using. This should not be a domain account.
* Un-install Erlang and RabbitMQ
* Remove the `%AppData%\RabbitMQ` directory entirely
* Install Erlang 23.0.3 from this location - http://erlang.org/download/otp_versions_tree.html
* Install RabbitMQ. *NOTE:* during the installation, un-click the "RabbitMQ Service" checkbox. This will prevent the windows service from being installed and run
* Open the "RabbitMQ Command Prompt (sbin dir)" start menu item
* Run this command, save all output to a file, and attach the file to your response
    ```
    .\rabbitmq-server.bat
    ```

**Reply by mrssuper** [reply]
thank you, i will give it a try
